File: backend/model/article.go
package model

type Article struct {
	ArticleIdentifier
	ArticleData
}

type ArticleIdentifier struct {
	Id string `json:"id"`
}

type ArticleData struct {
	Type      string   `json:"type"`
	Title     string   `json:"title"`
	CreatedAt string   `json:"createdAt"`
	UpdatedAt string   `json:"updatedAt"`
	TagKeys   []string `json:"tagKeys"`
	Summary   string   `json:"summary"`
	Content   string   `json:"content"`
}

File: frontend/types/data/article.ts
export type Article =
  & ArticleIdentifier
  & ArticleData
  & ArticleExtra;

export type ArticleIdentifier = {
  id: string;
};

export type ArticleData = {
  title: string;
  tagKeys: string[];
  summary: string;
  content: string;
  thumbnailUrl: string;
};

export type ArticleExtra = {
  createdAt: string;
  updatedAt: string;
};

export type ArticleSearchQuery = {
  q: string;
  tag: string[];
  sort: string;
  order: string;
  take: number;
  skip: number;
}